src/pinnable_mapped_file.cpp

```diff
@@ -433,31 +433,25 @@ void pinnable_mapped_file::save_database_file(bool flush /* = true */) {
    std::cerr << "CHAINBASE: Writing \"" << _database_name << "\" database file, complete." << '\n';
 }
 
-pinnable_mapped_file::pinnable_mapped_file(pinnable_mapped_file&& o)  noexcept :
-   _mapped_file_lock(std::move(o._mapped_file_lock)),
-   _data_file_path(std::move(o._data_file_path)),
-   _database_name(std::move(o._database_name)),
-   _file_mapped_region(std::move(o._file_mapped_region))
+pinnable_mapped_file::pinnable_mapped_file(pinnable_mapped_file&& o) noexcept
 {
-   _segment_manager = o._segment_manager;
-   _writable = o._writable;
-   _non_file_mapped_mapping = o._non_file_mapped_mapping;
-   o._non_file_mapped_mapping = nullptr;
-   o._segment_manager = nullptr;
-   o._writable = false; //prevent dtor from doing anything interesting
+   // all members are correctly default-initialized, so we can just move into *this
+   *this = std::move(o);
```

Comment:
Wouldn't this leave items like `_segment_manager` and `_writable` still "live" from what is being moved from? The old move constructor nulls these out so the dtor doesn't do stuff to something that has been moved from, but now it seems like it would

Reply:
I believe this would execute `pinnable_mapped_file::operator=(pinnable_mapped_file&& o)`, which swaps all the members, so `o._segment_manager` would get the current value or *this which is `nullptr` as initialized in the member declaration.

Reply:
okay yeah that makes sense